File: Solutions/832_Flipping_an_Image/832_Flipping_an_Image.py
class Solution:
    def flipAndInvertImage(self, A):
        return [[1^q for q in row[::-1]] for row in A]


#Third fastest
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        result = []
        for row in A:
            new_row = []
            result.append(new_row)
            for x in reversed(row):
                new_row.append(int(not x))
        return result


#Second fastest
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:

        for i in range(len(A)): 
            A[i].reverse()
            for j in range(len(A[i])): 
                if A[i][j] == 0: 
                    A[i][j] = 1
                else: 
                    A[i][j] = 0
        return A

#Fastest
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        
        B = [[] * len(A)]
        for row in A:
            for j in range((len(row) + 1) // 2):
                row[j], row[~j] = row[~j] ^ 1, row[j] ^ 1
        
        return A

File: Solutions/832_Flipping_an_Image/832_Flipping_an_Image.js
//Third fastest
var flipAndInvertImage = function(A) {
    for (let j = 0; j < A.length; j++) {
        imax = A[j].length-1;   // 2
        ii = Math.round(A[j].length / 2); // 3/2 1.5 2; 4/2 2 1 
        for (let i = 0; i < ii; i++) {
            let pix = A[j][i];
            A[j][i] = A[j][imax-i]
            A[j][imax-i] = pix;
        }
        for (let i = 0; i <= imax; i++) {
            A[j][i]   = 1 - A[j][i];
        }
    }
    return A;
};

//Second fastest
var flipAndInvertImage = function(A) {
    let result = []
    result = A.map((row)=>{
        let newRow = row.reverse();
        return newRow.map((e)=>1-e);
    })
    return result
};

//Fastest
/**
 * @param {number[][]} A
 * @return {number[][]}
 */
var flipAndInvertImage = function(A) { 
    for(let i=0;i<A.length;i++){
        A[i] = A[i].reverse().map(a => a==1 ? 0 : 1)
    }
    return A
};